pipeline/notifications/check_message.py

```diff
@@ -3,23 +3,30 @@
 import prefect
 from prefect import task
 
-from pipeline.types import ChatData, NotificationInfo, User
+from pipeline.types import NotificationInfo, NotificationInput
 
 
 class NewMessageNotifier:
     def __init__(self):
         pass
 
-    def get_users_to_notify(self, chatdata: ChatData) -> List[User]:
+    def get_users_to_notify(
+        self, inp: NotificationInput
+    ) -> List[NotificationInfo]:
 
         # check if all users in match have a message
-        # if there are users who don't have a message, return a list of them for notification
+        # if there are users who don't have a message, return a list of NotificationInfo objs associated w/ them
         users_with_msg = set()
         users_to_notify = []
-        for msg in chatdata.messages:
-            if msg.from_user in chatdata.participants:
+        for msg in inp.chatdata.messages:
+            if msg.from_user in inp.chatdata.participants:
                 users_with_msg.add(msg.from_user)
-        for uid, user in chatdata.participants.items():
+        for uid, user in inp.chatdata.participants.items():
             if uid not in users_with_msg:
-                users_to_notify.append(user)
+                notif_info_obj = NotificationInfo(
+                    name=user.displayName,
+                    email=user.email,
+                    title=inp.chatdata.title,
+                )
```

Comment:
I **would** use Counter here since it's O(n) compared to sorted's O(nlog(n)). But seeing as it's highly unlikely that n would be large (n being the number of users in a given match, which realistically should be, at most, 4), I believe it's fine to forgo some efficency. Plus we won't have to include a dependency (i.e. "from collections import Counter")

Reply:
Counter wouldn't be necessary because there's nothing to count. That's why you use a set or a dictionary.

Do you need to return a list here? Can messengers be a set? The main reason to convert a set to a list is if order matters. For testing, order doesn't matter when comparing two sets. Maybe you need order when you get to writing the email for the notification, in that case, you could let the display layer handle ordering. It's not a big deal since group chats are 2-3 users and you'll only get this notification if someone didn't message the chat at all, so we're talking about a messagers having only 1-2 values.